File: Lab9_1/src/main/java/MatrixGenerator.java
import java.util.Random;

public class MatrixGenerator {
    private int row_size;
    private int column_size;
    private double maxValue;

    Random random;

    MatrixGenerator(int size, double maxValue) {
        this(size, size, maxValue);
    }

    MatrixGenerator(int row_size, int column_size, double maxValue) {
        this.row_size = row_size;
        this.column_size = column_size;
        this.maxValue = maxValue;

        random = new Random();
    }

    double[][] generate() {
        double[][] matrix = new double[row_size][column_size];

        for (int i = 0; i < row_size; i++) {
            matrix[i] = new double[column_size];
            for (int j = 0; j < column_size; j++) {
                matrix[i][j] = maxValue * random.nextDouble();
            }
        }

        return matrix;
    }
}

File: Lab9_3/Lab9_3/MatrixGenerator.cpp
#include "MatrixGenerator.h"

MatrixGenerator::MatrixGenerator(int size, double maxValue) :MatrixGenerator(size, size, maxValue)
{
}

MatrixGenerator::MatrixGenerator(int rowSize, int columnSize, double maxValue) : rowSize(rowSize), columnSize(columnSize), maxValue(maxValue)
{
}

double** MatrixGenerator::generate()
{
	double** matrix = new double* [rowSize];

	for (int i = 0; i < rowSize; i++) {
		matrix[i] = new double[columnSize];
		for (int j = 0; j < columnSize; j++) {
			std::uniform_real_distribution<double> unif(0.0, maxValue);
			std::default_random_engine re;
			matrix[i][j] = unif(re);
		}
	}
	return matrix;
}
